### ISSUE TITLE
some erroneous expressions will not emit an error

### ISSUE THREAD

**Issue by danlangford** [Bernardo-MG/dice-notation-java#31]
so the expression `5dz` will throw an ANTRL `NoViableAltException` which is wrapped into a `IllegalStateException` however an expression like `6d6y` will not throw an exception that i can handle. It seems that some error is thrown at some point in ANTLR because the logs contain 

    line 1:3 token recognition error at: 'y'

however there is no simple way for me to capture that error.

**Comment by Bernardo-MG** [comment]
I'll check this as soon as I can. But I suspect the problem is that the second case is an error which ANTLR can fix, part of the expression matches a rule and just stops at the first failure, returning all it could parse.

The first example does not match any rule, so it will throw an exception.

**Comment by Bernardo-MG** [comment]
I found the problem, the lexer needed an error listener too:

```
lexer = new DiceNotationLexer(stream);
lexer.addErrorListener(new DefaultErrorListener());
```

This will be fixed for the next version, I'll try to get it done this month, as I want to add a few other fixes.

**Comment by Bernardo-MG** [comment]
I've just published version 1.3.1 with that change. It may take a short time for Maven Central to allow it being downloaded.

Try it and tell if it works.

**Comment by danlangford** [comment]
yes i am able to catch the exception now and communicate poor syntax to the user. thank you so much
